lib/express_validator.js

```diff
@@ -1,262 +1,335 @@
+ * @return {object[]}             array of errors
+ */
 
-          if (!isCorrect) {
-            errorHandler(failMsg || 'Invalid value');
-          }
 
-          return methods;
-        };
-      });
+/**
+ * Validates and handles errors, return instance of itself to allow for chaining
+ *
+ * @method makeValidator
+ * @param  {string}          methodName
+ * @param  {object}          container
+ * @return {function}
+ */
 
-      methods['notEmpty'] = function() {
-        return methods.isLength(1);
-      };
+function makeValidator(methodName, container) {
+  return function() {
+    var args = [];
+    args.push(this.value);
+    args = args.concat(Array.prototype.slice.call(arguments));
 
-      methods['len'] = function() {
-        return methods.isLength.apply(methods.isLength, Array.prototype.slice.call(arguments));
-      };
+    var isValid =  (this.skipValidating) ? true : container[methodName].apply(container, args);
 
-      methods['optional'] = function() {
-        if (value !== undefined) {
-          return methods;
-        }
+    if(!isValid) {
+      var error = this.errorFormatter(this.param, this.failMsg, this.value);
+      this.validationErrors.push(error);
+      this.req._validationErrors.push(error);
+    }
 
-        var dummyMethods = [];
-        for (var methodName in methods) {
-          dummyMethods[methodName] = function() { return dummyMethods; };
-        }
-        return dummyMethods;
-      };
+    return this;
+  };
+}
 
-      return methods;
-    };
-  }
-  return function(req, res, next) {
+/**
+ * Sanitizes and sets sanitized value on the request, then return instance of itself to allow for chaining
+ *
+ * @method makeSanitizer
+ * @param  {string}          methodName
+ * @param  {object}          container
+ * @return {function}
+ */
 
-    req.updateParam = function(name, value) {
-      // route params like /user/:id
-      if (this.params && this.params.hasOwnProperty(name) &&
-        undefined !== this.params[name]) {
-        return this.params[name] = value;
-      }
-      // query string params
-      if (undefined !== this.query[name]) {
-        return this.query[name] = value;
+function makeSanitizer(methodName, container) {
+  return function() {
+    var _arguments = arguments;
+    this.values.forEach(function(value, i) {
+      if(value) {
+        var args = [value];
+        args = args.concat(Array.prototype.slice.call(_arguments));
+        var result = container[methodName].apply(container, args);
+
+        set(this.req[this.locations[i]], this.param, result);
+        this.values[i] = result;
       }
-      // request body params via connect.bodyParser
-      if (this.body && undefined !== this.body[name]) {
-        return this.body[name] = value;
-      }
-      return false;
-    };
+    }.bind(this));
 
-    req.check = checkParam(req, function(item) {
-      return param(req, item);
-    });
+    return this;
```

Comment:
If this is the only thing, yeah, let's change it back to `value` just to make this a frictionless update. If someone specifically needs multiple sanitizers on one value, I would suggest they use multiple lines or create a custom sanitizer. Obviously, update docs/unit tests related to this.